tap_mysql/connection.py

```diff
@@ -21,7 +23,28 @@
                       max_tries=1,
                       factor=2)
 def connect_with_backoff(connection):
-    connection.connect()
+    try:
+        connection.connect()
+    except pymysql.err.OperationalError as e:
+        message = str(e)
+        if 'Access denied for user' in message:
+            raise SymonException('The username and password provided are incorrect. Please try again.', 'odbc.AuthenticationFailed')
+        if 'nodename nor servname provided, or not known' in message:
+            raise SymonException(f'The host "{connection.host}" was not found. Please check the host name and try again.', 'odbc.HostNotFound')
+        if "Can't connect to MySQL server" in message:
```

Comment:
there could be three sub-errors for this error: timeout, network unreachable, connection refused. Time out handled separately. Other two errors do not indicate host is wrong 100% (could be due to wrong configuration on database server/ server down)